stages/test/test_erofs.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/python3
+
+import os.path
+import subprocess
+from unittest import mock
+
+import pytest
+
+import osbuild.meta
+from osbuild.testutil.imports import import_module_from_path
+
+
+def make_fake_input_tree(tmpdir, fake_content: dict) -> str:
+    basedir = os.path.join(tmpdir, "tree")
+    for path, content in fake_content.items():
+        dirp, name = os.path.split(os.path.join(basedir, path.lstrip("/")))
+        os.makedirs(dirp, exist_ok=True)
+        with open(os.path.join(dirp, name), "w", encoding="utf-8") as fp:
+            fp.write(content)
+    return basedir
+
+
+def have_erofs() -> bool:
+    try:
+        r = subprocess.run(
+            ["mkfs.erofs", "--help"], check=False, stderr=subprocess.DEVNULL)
+    except FileNotFoundError:
+        return False
+    return r.returncode == 0
+
+
+@pytest.mark.skipif(not have_erofs(), reason="no mkfs.erofs")
+def test_erofs_integration(tmp_path):
+    fake_input_tree = make_fake_input_tree(tmp_path, {
+        "/file-in-root.txt": "other content",
+        "/subdir/file-in-subdir.txt": "subdir content",
+    })
+    inputs = {
+        "tree": {
+            "path": fake_input_tree,
+        }
+    }
+    stage_path = os.path.join(os.path.dirname(__file__), "../org.osbuild.erofs")
+    stage = import_module_from_path("erofs_stage", stage_path)
+    filename = "some-file.img"
+    options = {
+        "filename": filename,
+    }
+
+    stage.main(inputs, tmp_path, options)
```

Comment:
The integration test could also take similar inputs as the mocked one and double check using the "dump.erofs" that the expected options are part of the dump.erofs output

Reply:
This is now done